shell/renderer/api/context_bridge/object_cache.cc

```diff
@@ -21,7 +21,15 @@ ObjectCachePairNode::ObjectCachePairNode(ObjectCachePair&& pair) {
 ObjectCachePairNode::~ObjectCachePairNode() = default;
 
 ObjectCache::ObjectCache() {}
-ObjectCache::~ObjectCache() = default;
+ObjectCache::~ObjectCache() {
+  for (const auto& pair : proxy_map_) {
+    while (!pair.second.empty()) {
+      ObjectCachePairNode* node = pair.second.head()->value();
+      node->RemoveFromList();
+      delete node;
+    }
+  }
+}
```

Comment:
This is a little larger scope, but -- what is the benefit of using base::LinkedList here instead of std::list, which is simpler / doesn't require a ObjectCachePairNode / doesn't have these gotchas?

According to the docs, the  main reason to use base::LinkedList are for performance reasons of (1) removal is O(1) rather than O(n), and (2) insertion with base::LinkedList<T> never requires heap allocations. But removal never called except in the destructor, and the items we insert are heap-allocated ObjectCachePairNodes, so do either of those advantages apply here?

Reply:
Agree in general, I want to land this fix then I'll follow it up to see if we can remove our usage of `LinkedList`.  I can't remember exactly why we used it originally, but this code has been through several refactors and it's possible our usage of LinkedList is no longer required.